### PR TITLE
chore: convert configuration-manager to TS

### PR DESCRIPTION
**Pull request by jhockett**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/extensions/amplify-helpers/make-id.ts [hunk 1]
```diff
@@ -1,11 +1,8 @@
-export function makeId(n) {
-  if (!n) {
-    n = 5;
-  }
+export function makeId(n: number = 5) {
   let text = '';
   const possible = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
 
-  for (let i = 0; i < n; i += 1) {
+  for (let i = 0; i < n; ++i) {
```
Comment (akshbhu) [inline packages/amplify-cli/src/extensions/amplify-helpers/make-id.ts:5]: Will this code face collisions in ID when n is small?

Comment (jhockett) [reply packages/amplify-cli/src/extensions/amplify-helpers/make-id.ts:5]: Possibly. I just edited this file to fix the types for the time being.

File: packages/amplify-cli-core/src/index.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ export type $TSContext = {
   usageData: $TSAny;
   runtime: $TSAny;
   pluginPlatform: IPluginPlatform;
+  newUserInfo?: $TSAny;
```
Comment (nikhname) [inline packages/amplify-cli-core/src/index.ts:37]: Can this be more strongly typed?

Comment (jhockett) [reply packages/amplify-cli-core/src/index.ts:37]: Not with ease. I don't fully know how it's being used throughout the CLI. This was added to fix TS compilation. We should refactor the functions that set this value to return the data directly instead mutating `context`, but that was out of scope for this PR.

**Review by nikhname — APPROVED** [review]
LGTM!

**Pull request merged**